src/java.base/share/classes/java/util/zip/ZipInputStream.java

```diff
@@ -172,13 +173,16 @@ public void closeEntry() throws IOException {
     }
 
     /**
-     * Returns 0 after EOF has reached for the current entry data,
-     * otherwise always return 1.
+     * Returns 0 when end of stream is detected for the current ZIP entry or
+     * {@link #closeEntry()} has been called, otherwise always return 1.
+     * I think it will need to say that input stream for the current ZIP entry has been read to end of stream, or the
+     * ZIP entry has been closed with closeEntry
```

Comment:
A comment from the PR has been pasted here :-)
Ignoring that, the updated description for the "available" method looks okay.

Reply:
> A comment from the PR has been pasted here :-)
> Ignoring that, the updated description for the "available" method looks okay but might be clearer to say that closeEntry has been called on the current entry.

whoops, fixed :-)

Reply:
I just went and added the extra wording regarding the "current entry".    I did not add it originally because the verbiage links to closeEntry which indicates that it closes the current entry and thought it might be overkill